spec/services/make_an_offer_spec.rb

```diff
@@ -38,4 +38,37 @@
       expect(decision).not_to be_valid
     end
   end
+
+  describe 'decline by default' do
+    def create_application
+      application_form = create :completed_application_form
+      create(
+        :application_choice,
+        application_form: application_form,
+        status: 'awaiting_provider_decision',
+        edit_by: 2.business_days.ago,
+        )
+    end
+
+    it 'sets the `offered_at` date' do
+      application_choice = create_application
+
+      Timecop.freeze do
+        MakeAnOffer.new(application_choice: application_choice).save
+        expect(application_choice.offered_at).to eq(Time.now)
+      end
+    end
+
+    it 'calls SetDeclineByDefault service' do
```

Comment:
Instead of the stubbing and instances, can we check if the attributes are updated on the form? This does a lot of testing of internals.

Reply:
There are 2 reasons:
- SetDeclineByDefault is already extensively tested on `set_decline_by_default_spec.rb`
- the `MakeAnOffer` spec is already using a lot of stubs (`MakeAnOffer`), changing the style of the test would require rewriting all the other tests

Reply:
You don’t have to test the exact values (which would dupe the existing tests), just make sure the attributes on the form are updated.

Reply:
good call @tijmenb, I've changed it!